src/js/helpers/Util.js

```diff
@@ -105,8 +105,8 @@ var Util = {
     } else if (this.isArray(obj)) {
       copy = obj.slice(); // shallow copy
 
-      copy.forEach((value, index) => {
-        copy[index] = this.deepCopy(value);
+      Object.keys(copy).forEach((key) => {
```

Comment:
This is not right. If you run `Object.keys` on an array it returns the indices present at in the array. If the following snippet gets executed only ["10"] will be returned by `Object.keys`
```JS
var x = new Array(25);
x[10] = "example";
console.log(Object.keys(x));
```
This is also the awaited function from [Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)
> The Object.keys() method returns an array of a given object's own enumerable properties, in the same order as that provided by a for...in loop (the difference being that a for-in loop enumerates properties in the prototype chain as well).